### PR TITLE
Feature - Batch To Intacct - Account Grouping

### PR DESCRIPTION
**Pull request by guymaness2**
##### What does the change add or fix?

Added a block setting to control grouping of accounts in the resulting Intacct Journal Entry or Other Receipt.

**New Settings:**
Batch to Journal and Batches to Journal Block
* GL Account Grouping - Determines if debit and/or credit lines should be grouped and summed by GL account in the resulting Intacct Journal Entry or Other Receipt.
---------

##### What does the change add or fix in a succinct statement that will be read by clients?

* Added block settings to group accounts in resulting Intacct Journal Entries or Other Receipts. 

---------

##### Who reported, requested, or paid for the change?

The Father's House

---------

##### Does this update or add options to the block UI?
  
New block setting for Batch To Journal and Batches To Journal blocks:  
image  

Example 1 (Default grouping - Legacy):
image  
  
Example 2 (Debit Lines Only):  
image
  
Example 3 (Debit And Credit Lines):  
image  

---------

##### What files does it affect?

* Intacct/BatchToJournal.ascx.cs
* Intacct/BatchesToJournal.ascx.cs  

---------

##### Is it a breaking change for other versions/clients?

Has dependency on https://github.com/KingdomFirst/RockAssemblies/pull/129

### CODE REVIEW

File: Intacct/BatchToJournal.ascx.cs [hunk 1]
```diff
@@ -156,9 +156,19 @@ namespace RockWeb.Plugins.rocks_kfs.Intacct
         ListSource = "JournalEntry,OtherReceipt",
         DefaultValue = "JournalEntry",
         Category = "Configuration",
-        Order = 6,
+        Order = 13,
         Key = AttributeKey.ExportMode )]
 
+    [EnumField(
+        "GL Account Grouping",
+        Description = "Determines if debit and/or credit lines should be grouped and summed by GL account in the export file. NOTE: Unique ProjectIds, DepartmentIds, LocationIds, etc. may result in multiple lines even if account is grouped. Also, DebitLinesOnly mode is unsupported for Other Receipts.",
```
Comment (guymaness2) [inline Intacct/BatchToJournal.ascx.cs:164]:
```suggestion
        Description = "Determines if debit and/or credit lines should be grouped and summed by GL account in the resulting Intacct Journal Entry or Other Receipt. NOTE: Unique ProjectIds, DepartmentIds, LocationIds, etc. may result in multiple lines even if account is grouped. Also, DebitLinesOnly mode is unsupported for Other Receipts.",
```

File: Intacct/BatchesToJournal.ascx.cs [hunk 2]
```diff
@@ -188,6 +188,16 @@ namespace RockWeb.Plugins.rocks_kfs.Intacct
         Order = 15,
         Key = AttributeKey.ExportMode )]
 
+    [EnumField(
+        "GL Account Grouping",
+        Description = "Determines if debit and/or credit lines should be grouped and summed by GL account in the export file. NOTE: Unique ProjectIds, DepartmentIds, LocationIds, etc. may result in multiple lines even if account is grouped. Also, DebitLinesOnly mode is unsupported for Other Receipts.",
```
Comment (guymaness2) [inline Intacct/BatchesToJournal.ascx.cs:193]:
```suggestion
        Description = "Determines if debit and/or credit lines should be grouped and summed by GL account in the resulting Intacct Journal Entry or Other Receipt. NOTE: Unique ProjectIds, DepartmentIds, LocationIds, etc. may result in multiple lines even if account is grouped. Also, DebitLinesOnly mode is unsupported for Other Receipts.",
```

File: Intacct/ReadMe.md [hunk 3]
```diff
@@ -136,37 +141,25 @@ You will need to configure the Batch to Journal block settings.
...
+| <span style="width: 3em; height: 3em; line-height: 3em; background: #d21919; border-radius: 100%; color: white; text-align: center; display: inline-block;">16</span> | **GL Account Grouping** Determines if debit and/or credit lines should be grouped and summed as follows in the export file:<ul><li>**Debit Accounts**: Company, Region, Super Fund, Cost Center Debit Number, Debit Account, Debit Account Sub, Fund Number, Project, Transaction Fee Account, and Location</li><li>**Credit Accounts**: Company, Region, Department, Super Fund, Cost Center Credit Number, Revenue Account, Revenue Account Sub, Fund Number, Project, and Location</li><li>**Financial Account Grouping**: Both credit and debit lines are grouped by Rock Financial Account, Project, and Transaction Fee Account.</li></ul> |
```
Comment (guymaness2) [inline Intacct/ReadMe.md:161]:
```suggestion
| <span style="width: 3em; height: 3em; line-height: 3em; background: #d21919; border-radius: 100%; color: white; text-align: center; display: inline-block;">16</span> | **GL Account Grouping** Determines if debit and/or credit lines should be grouped and summed as follows in the resulting Intacct Journal Entry or Other Receipt:<ul><li>**Debit Accounts**: Company, Region, Super Fund, Cost Center Debit Number, Debit Account, Debit Account Sub, Fund Number, Project, Transaction Fee Account, and Location</li><li>**Credit Accounts**: Company, Region, Department, Super Fund, Cost Center Credit Number, Revenue Account, Revenue Account Sub, Fund Number, Project, and Location</li><li>**Financial Account Grouping**: Both credit and debit lines are grouped by Rock Financial Account, Project, and Transaction Fee Account.</li></ul> |
```

File: Intacct/ReadMe.md [hunk 4]
```diff
@@ -233,11 +214,11 @@ If your financial gateway reports transaction fees to Rock in their transaction
...
-    
-    Default Fee Account: Default account number for transaction fees.
-```
+| | |
+| --- | ---- |
+| **Gateway Fee Processing** How should the Intacct Export plugin process transaction fees? DEFAULT: No special handling of transaction fees will be performed. NET DEBIT: Add credit entries for any transaction fees and use net amount (amount - transaction fees) for debit account entries. GROSS DEBIT: Debit account entries are left untouched (gross) and new debit and credit entries will be added for any transaction fees. NOTE: Both Net Debit and Gross Debit require a Fee Account attribute be set on either the financial gateway or financial account. |
+| **Default Fee Account** TDefault account number for transaction fees. |
```
Comment (nateh777) [inline Intacct/ReadMe.md:220]:
```suggestion
| **Default Fee Account** Default account number for transaction fees. |
```

File: Intacct/ReadMe.md [hunk 5]
```diff
@@ -178,45 +171,33 @@ The Batches to Journal block was added to a new page under your Finances heading
...
+| <span style="width: 3em; height: 3em; line-height: 3em; background: #d21919; border-radius: 100%; color: white; text-align: center; display: inline-block;">18</span> | **GL Account Grouping** Determines if debit and/or credit lines should be grouped and summed as follows in the export file:<ul><li>**Debit Accounts**: Company, Region, Super Fund, Cost Center Debit Number, Debit Account, Debit Account Sub, Fund Number, Project, Transaction Fee Account, and Location</li><li>**Credit Accounts**: Company, Region, Department, Super Fund, Cost Center Credit Number, Revenue Account, Revenue Account Sub, Fund Number, Project, and Location</li><li>**Financial Account Grouping**: Both credit and debit lines are grouped by Rock Financial Account, Project, and Transaction Fee Account.</li></ul> |
```
Comment (guymaness2) [inline Intacct/ReadMe.md:200]:
```suggestion
| <span style="width: 3em; height: 3em; line-height: 3em; background: #d21919; border-radius: 100%; color: white; text-align: center; display: inline-block;">18</span> | **GL Account Grouping** Determines if debit and/or credit lines should be grouped and summed as follows in the resulting Intacct Journal Entry or Other Receipt:<ul><li>**Debit Accounts**: Company, Region, Super Fund, Cost Center Debit Number, Debit Account, Debit Account Sub, Fund Number, Project, Transaction Fee Account, and Location</li><li>**Credit Accounts**: Company, Region, Department, Super Fund, Cost Center Credit Number, Revenue Account, Revenue Account Sub, Fund Number, Project, and Location</li><li>**Financial Account Grouping**: Both credit and debit lines are grouped by Rock Financial Account, Project, and Transaction Fee Account.</li></ul> |
```

**Comment by guymaness2** [comment]
@nateh777 Screenshots added, suggestions applied, extra table column removed, pdf regenerated. I think we are go for merge.

**Review by nateh777 — APPROVED** [review]
@guymaness2 can you check on the page break on page 14 of the pdf? Can we just remove that page break so the table will at least show where the image gets cutoff? Or will it push it anyway? Just curious as that page just has a cutoff image, or if we can resize the image down slightly so it doesn't get cutoff? I will go ahead and merge so you can package but if you can try editing the pdf for that?

**Pull request merged**